### DISCUSSION TITLE
部分模型的视线追踪问题

### DISCUSSION THREAD

**Discussion by GardenHamster** [guansss/pixi-live2d-display#143] [Q&A]
有一个模型其实在游戏里面是没有视线跟踪效果的(就是模型的视觉会跟着玩家鼠标移动的那种)
https://cdn.jsdelivr.net/gh/GardenHamster/live2d-web/model/Houkai-Gakuen2/hongyexiangshui/1833_cn.model3.json
但是导入到Live2DViewerEx中是有鼠标追踪效果的
请教一下有没有什么办法在web上也能实现类似的效果

**Comment by guansss** [comment]
这个模型的 param id 比较特别，比如正常模型的 `ParamEyeBallX` 在这里是 `PARAM_EYE_BALL_X`，所以把需要用到的 id 替换一下就好了：

```js
const model = await Live2DModel.from(modelURL);

model.internalModel.idParamEyeBallX = "PARAM_EYE_BALL_X";
model.internalModel.idParamEyeBallY = "PARAM_EYE_BALL_Y";
model.internalModel.idParamAngleX = "PARAM_ANGLE_X";
model.internalModel.idParamAngleY = "PARAM_ANGLE_Y";
model.internalModel.idParamAngleZ = "PARAM_ANGLE_Z";
```

不清楚为什么 Live2DViewerEx 可以识别这种 id，可能专门做了适配吧

**Comment by GardenHamster** [comment]
明白了,谢谢佬

**Comment by GardenHamster** [comment]
我翻了下 Live2DViewerEx  的文档，确实是按照SDK做了适配的

http://live2d.pavostudio.com/doc/zh-cn/exstudio/live2d-editor/#_18

![image](https://github.com/guansss/pixi-live2d-display/assets/89188316/c91bfef4-67f7-4ff6-a54f-cc86aeef5b4f)

另外想问下这个项目支持自动眨眼吗，我发现这个模型同样是没有自动眨眼的，会不会也跟参数有关

但是我在 model.internalMode l里没有找到相关的参数设置了

![image](https://github.com/guansss/pixi-live2d-display/assets/89188316/2ca7d8c0-34a1-4cef-8eca-69a98c9944a3)

**Comment by guansss** [comment]
确实，应该是把两种参数共用了

> 另外想问下这个项目支持自动眨眼吗，我发现这个模型同样是没有自动眨眼的，会不会也跟参数有关

有的，但只有在未播放 motion 的情况下才会触发，因为一般 motion 都是带有眼部动画的（包括眨眼），如果再加上自动眨眼会冲突
